### ISSUE TITLE
Missing http request header when sending location.

### ISSUE THREAD

**Issue by momo02** [transistorsoft/react-native-background-geolocation#1228]
* Plugin version: 3.9.3
* Platform: Android
* OS version: 11.0
* Device manufacturer / model: Using the Android Emulator
* React Native version (`react-native -v`): 0.63.3
* Plugin config
```javascript <!-- Syntax-highlighting:  paste your code below -->

 const loginToken = await getLoginToken();
 console.log("loginToken :::::: ", loginToken); // login token is printed well.

 BackgroundGeolocation.ready({
                reset: true,
                // Geolocation Config
                desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
                distanceFilter: 10,
                // Activity Recognition
                stopTimeout: 1,
                // Application config
                debug: false, // <-- enable this hear sounds for background-geolocation life-cycle.
                logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
                stopOnTerminate: false,   // <-- Allow the background-service to continue tracking when user closes the app.
                startOnBoot: true,        // <-- Auto start tracking when device is powered-up.
                // HTTP / SQLite config
                url: `${API_URL}/pty/lct`,
                headers: {              // <-- Optional HTTP headers
                    'Authorization': loginToken,
                },
                httpRootProperty: 'lct',
                locationTemplate: '{"lat":<%= latitude %>,"lon":<%= longitude %>,"speed":<%= speed %>}',
                batchSync: false,       // <-- [Default: false] Set true to sync locations to server in a single HTTP request.
                autoSync: true,         // <-- [Default: true] Set true to sync each location to server as it arrives.
                autoSyncThreshold: 5,
                httpTimeout: 6000,
                heartbeatInterval: 60 * 1,
                stopOnStationary: false,
                enableHeadless: true
            }, (state) => {
                console.log("- BackgroundGeolocation is configured and ready: ", state.enabled);
                if (!state.enabled) {
                    ////
                    // 3. Start tracking!
                    //
                    BackgroundGeolocation.start(function () {
                        console.log("- Start success");
                    });
                }
            });

        }
```

## Expected Behavior
Send location information to server with jwt accessToken.

## Actual Behavior

Before calling the BackgroundGeolocation.ready function, the console log prints jwt tokens starting with "Safety-Tour".
also 200 response is returned even if the request is sent using postman with the printed login Token.
![image](https://user-images.githubusercontent.com/23231110/105942446-996da400-60a2-11eb-8956-de8f9345521c.png)

However, onHttp response returns an error of 401.
response log : {responseText: "{"message":"권한이 없습니다.","error":"401","status":401}", status: 401, success: false}

A 401 privilege error occurs if the token in the http header is invalid.
Below is the log on the server side. 

2021-01-27 10:18:37 [WARN ] [JwtInterceptor.java]preHandle(72) : JWT Token does not begin with Safety-Tour  String

We tried the authorization option instead of the header option, ( token is not refreshed, so all options are not used)
```javascript <!-- Syntax-highlighting:  paste your code below -->
                authorization: {
                    strategy: "JWT",
                    accessToken: loginToken
                    //refreshToken: loginToken
                    // refreshUrl: "https://auth.your.server.com/tokens",
                    // refreshPayload: {
                    //     the_refresh_token_field_name: "{refreshToken}"
                    // },
                    // expires: myToken.expiresAt
                },
```
         
but it was the same. Did I make something wrong? Please check...

I also have one more question.

Is onHttp callback function only called when location is updated..?
I look forward to hearing from you. Thank you.

## Steps to Reproduce

1. Use configuration above and generate location update event to check on Http callback function

## Debug logs

<details><summary>Logs</summary>

``` <!-- Syntax highlighting:  DO NOT REMOVE -->
[onHttp] FAILURE:  {responseText: "{"message":"권한이 없습니다.","error":"401","status":401}", status: 401, success: false}
```

</details>

**Comment by christocracy** [comment]
Are you observing the Android logs with `$ adb logcat`?  See Wiki *Debugging*.

**Comment by momo02** [comment]
> Are you observing the Android logs with `$ adb logcat`? See Wiki _Debugging_.

I figured out the cause of the problem.

BackgroundGeolocation.ready() was sometimes called before logging in. Therefore, the login token was sometimes null.
Also, I didn't know that config has been remembering it ever since I installed the app.

On our server side, tokens are only refreshed when re-logged in, and api requesting a token refresh is not provided.

Should we always call the ready function after login, and set false the reset option in Config for re-login?
